### PR TITLE
imxrt: Fix sleep config, addded missing bariers

### PR DESCRIPTION
**Pull request by agkaminski**
JIRA: RTOS-101

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imxrt117x).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/241).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by nalajcie** [comment]
Just a thing I've seen in e-mail - commit message has a typo (too many d's in `addded`). My preference is to use imperative mood in commit messages, but we don't have any guidelines regarding that ;).

File: hal/armv7m/imxrt/117x/imxrt.c [hunk 1]
```diff
@@ -562,33 +592,51 @@ void _imxrt_init(void)
...
-		*(imxrt_common.aips[i] + aipstz_opacr) &= ~0x44444444;
-		*(imxrt_common.aips[i] + aipstz_opacr1) &= ~0x44444444;
-		*(imxrt_common.aips[i] + aipstz_opacr2) &= ~0x44444444;
-		*(imxrt_common.aips[i] + aipstz_opacr3) &= ~0x44444444;
-		*(imxrt_common.aips[i] + aipstz_opacr4) &= ~0x44444444;
-	}
-*/
+	/* Enable UsageFault, BusFault and MemManage exceptions */
+	*(imxrt_common.scb + scb_shcsr) |= (1 << 16) | (1 << 17) | (1 << 18);
+	imxrt_dataBarrier();
```
Comment (gerard5) [inline hal/armv7m/imxrt/117x/imxrt.c:641]: Additionally to enabled exceptions I would enable [DIV_0_TRP](https://developer.arm.com/documentation/dui0646/c/cortex-m7-peripherals/system-control-block/configuration-and-control-register) in SCB->CCR , what do You think ?

File: hal/armv7m/imxrt/10xx/imxrt.c [hunk 2]
```diff
@@ -1871,6 +1875,13 @@ void _imxrt_init(void)
...
+	/* Enable SEVEONPEND bit */
+	*(imxrt_common.scb + scb_scr) |= 1 << 4;
+	imxrt_dataBarrier();
+
+	/* Disable deep sleep */
+	*(imxrt_common.scb + scb_scr) &= ~(1 << 2);
+	imxrt_dataBarrier();
 
 	/* Store reset flags and then clean them */
 	imxrt_common.resetFlags = *(imxrt_common.src + src_srsr) & 0x1f;
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply hal/armv7m/imxrt/10xx/imxrt.c]: Done

File: hal/armv7m/imxrt/117x/imxrt.c (hunk 1 above)
Comment (agkaminski) [reply hal/armv7m/imxrt/117x/imxrt.c:634]: Maybe in the future, I want to reduce number of changes to the minimum here

File: hal/armv7m/imxrt/117x/_init.S [hunk 3]
```diff
@@ -181,34 +181,38 @@ _init_vectors:
 .type _start, %function
 _start:
 	cpsid if
-
+#if 0
```
Comment (HBuczynski) [inline hal/armv7m/imxrt/117x/_init.S:184]: Why have you switched off `FLEX RAM` configuration? Comment will be helpful :)

File: hal/armv7m/imxrt/117x/imxrt.c [hunk 4]
```diff
@@ -562,6 +594,20 @@ void _imxrt_init(void)
...
 		*(imxrt_common.stk + stk_ctrl) &= ~1;
 
+	/* Disable deep sleep */
+	*(imxrt_common.scb + scb_scr) &= ~(1 << 2);
+	imxrt_dataBarrier();
+
+	/* HACK - temporary fix for crash when booting from FlexSPI */
+	_imxrt_setDirectLPCG(pctl_lpcg_semc, 0);
+
+	/* Disable USB cache (set by bootrom) */
```
Comment (HBuczynski) [inline hal/armv7m/imxrt/117x/imxrt.c:604]: Isn't it needed for USB device controller?

File: hal/armv7m/imxrt/117x/imxrt.c [hunk 5]
```diff
@@ -332,8 +340,7 @@ u8 _imxrt_nvicGetPriority(s8 irqn)
 void _imxrt_nvicSystemReset(void)
 {
 	*(imxrt_common.scb + scb_aircr) = ((0x5fa << 16) | (*(imxrt_common.scb + scb_aircr) & (0x700)) | (1 << 0x02));
-
-	__asm__ volatile ("dsb");
```
Comment (HBuczynski) [inline hal/armv7m/imxrt/117x/imxrt.c:336]: The same `__asm__` code is in:
https://github.com/phoenix-rtos/plo/blob/acd42a0b7c03c3c17fc7d90fb9db3831e1af3fde/hal/armv7m/imxrt/10xx/imxrt.c#L1553

Can you change it there as well?

File: hal/armv7m/imxrt/117x/imxrt.c (hunk 4 above)
Comment (agkaminski) [reply hal/armv7m/imxrt/117x/imxrt.c:604]: This is something that bootrom set and haven't cleaned after itself. It is set after boot from FlexSPI, but reset after boot from serial downloader. It's done in NXP's SDK.

File: hal/armv7m/imxrt/117x/_init.S (hunk 3 above)
Comment (agkaminski) [reply hal/armv7m/imxrt/117x/_init.S]: Done

**Pull request merged**